Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM nimlang/nim:latest AS builder
+
+COPY src/runner.nim /test-runner/
+WORKDIR /test-runner/
+RUN nim c runner.nim
+
+FROM nimlang/nim:latest-alpine-slim
+RUN apk add --no-cache pcre
+WORKDIR /opt/test-runner/
+COPY --from=builder /test-runner/runner bin/
+COPY . .
+RUN chmod +x bin/run.sh
```

Comment:
I think you could set the executable bit on the file on the file system itself (outside Docker). Then you would not need this, unless you don't want the `bin/run.sh` file to be executable outside the Docker container.

Reply:
Good point. I removed the `chmod` - the executable bit was already set.